lib/fluent/plugin/out_elasticsearch.rb

```diff
@@ -366,7 +367,13 @@ def create_time_parser
           Proc.new { |value| DateTime.strptime(value, @time_key_format) }
```

Comment:
I think you're gonna need similar handling here as well. While I was hunting down this problem, I tried providing a `time_key_format` of `%s.%N`, but that also failed for the same reason.

Reply:
I didn't use the `strptime` gem, but I imagine it'll need similar checking (line 361).

Reply:
I wish you showed me your Fluentd configuration and reproducible steps.
Just commenting not work for me is totally useless to solve issue, thanks!!

Reply:
The only relevant variables here are `@time_key_format` and the value that gets passed into the `Proc` object, so I thought the `time_key_format` configuration option that I provided was enough. The rest of any config file I could provide wouldn't matter.

Since there are three code paths in this function and you're only covering one, I think it would make more sense to pull lines 372 and 373 into a separate function, maybe named `convert_numeric_time_to_string`. Then update the three `Proc` blocks in the `create_time_parser` function to:
```
Proc.new { |value|
  if value.is_a?(Numeric)
    value = convert_numeric_time_to_string(value)
  end
  ...
```

Reply:
Yeah, it is reasonable for me.

Reply:
I've committed to follow your suggestion with a few improving.
Could you check again?